Library/Homebrew/cmd/uninstall.rb

```diff
@@ -64,6 +43,10 @@ def uninstall
   rescue MultipleVersionsInstalledError => e
     ofail e
     puts "Use `brew remove --force #{e.name}` to remove all versions."
+  ensure
+    # If we delete Cellar/newname, then Cellar/oldname symlink
+    # can become broken and we have to remove it.
+    HOMEBREW_CELLAR.children.select(&:symlink?).reject(&:resolved_path_exists?).each(&:unlink)
```

Comment:
I think it's actually worth to do this optimization here as one can install formulae as many as possible. Thanks for the advice here.